src/NSubstitute/Proxies/DelegateProxy/DelegateProxyFactory.cs

```diff
@@ -67,31 +68,95 @@ private Type GenerateDelegateContainerInterface(Type delegateType)
                 delegateTypeName,
                 typeSuffixCounter.ToString(CultureInfo.InvariantCulture));
 
-            var typeBuilder = _castleObjectProxyFactory.DefineDynamicType(
-                typeName,
-                TypeAttributes.Abstract | TypeAttributes.Interface | TypeAttributes.Public);
+            return _castleObjectProxyFactory.DefineDynamicType(moduleBuilder =>
+            {
+                var typeBuilder = moduleBuilder.DefineType(
+                    typeName,
+                    TypeAttributes.Abstract | TypeAttributes.Interface | TypeAttributes.Public);
+
+                // Notice, we don't copy the custom modifiers here.
```

Comment:
This code is really nicely documented. 👍 (code comments sometimes get a lot of hate, so I wanted to give good credit where it is due 😂 )